src/CsBot/IrcBotService.cs

```diff
@@ -0,0 +1,352 @@
+					Writer.WriteLine ($"PRIVMSG mattermost LOGIN {Settings.Nick} {Settings.Password}");
+					Writer.Flush ();
+				} else if (inputLine.Contains (":Nickname is already in use.")) {
+					Console.WriteLine ("Reopening with _ nick.");
+
+					Writer.Close ();
+					Reader.Close ();
+					ircTcpClient.Close ();
+					ircTcpClient = new TcpClient ();
+					ircTcpClient.Connect (Settings.Server, Settings.Port);
+					if (Settings.Secure) {
+						stream = new SslStream (ircTcpClient.GetStream (), true, ValidateServerCertificate);
+						var sslStream = (SslStream)stream;
+						sslStream.AuthenticateAsClient (Settings.Server);
+					} else {
+						stream = ircTcpClient.GetStream ();
+					}
+
+					Reader = new StreamReader ((Stream)stream);
+					Writer = new StreamWriter ((Stream)stream);
+					Console.WriteLine (Settings.User);
+					// Run PingSender thread
+					ping = new PingSender (this);
+					ping.Start ();
+					Writer.WriteLine (Settings.User);
+					Writer.Flush ();
+					Writer.WriteLine ($"NICK _{Settings.Nick}");
+					Console.WriteLine ($"NICK _{Settings.Nick}");
+					commandHandler = new CommandHandler (this);
+					isUnderscoreNick = true;
+				} else if (inputLine.Contains ("PRIVMSG") && (inputLine.Contains ("rock") || inputLine.Contains ("paper") || inputLine.Contains ("scissors"))) {
+					Console.WriteLine (inputLine);
+					addresser = inputLine.Substring (inputLine.IndexOf (":") + 1, inputLine.IndexOf ("!") - inputLine.IndexOf (":") - 1);
+					var choice = inputLine.Substring (inputLine.LastIndexOf (":") + 1);
+					commandHandler.DirectRoShamBo (choice);
+
+				} else if (inputLine.Contains ("PRIVMSG") && inputLine.Contains ($":{Settings.CommandStart}")) {
+					Console.WriteLine ($"PrivateMessage: {inputLine}");
+					addresser = inputLine.Substring (1, inputLine.IndexOf ("!") - 1);
+					var command = inputLine.Substring (inputLine.LastIndexOf ($":{Settings.CommandStart}"));
+					commandHandler.HandleMessage (command, addresser, addresser);
+				} else {
+					if (inputLine.Contains ("PRIVMSG") && inputLine.Contains ("!")) {
+						var userName = inputLine.Substring (1, inputLine.IndexOf ("!") - 1);
+						commandHandler.LastMessage (userName, inputLine, fromChannel);
+					}
+				}
+			}
+		}
+
+		async Task ObtainIrcConfig ()
+		{
+			// FIXME, the config stuff needs to be reviewed
+			_log.LogInformation ("Trying to pull config.");
+			_log.LogInformation ($"Settings file: {Constants.SettingsFile}");
+
+			string settingsJson;
+			if (_settingsFileLocation.StartsWith ("http"))
+				settingsJson = await DownloadIrcConfig (new Uri (_settingsFileLocation));
+			else
+				settingsJson = LoadLocalIrcConfig (Path.Combine (Directory.GetCurrentDirectory (), _settingsFileLocation));
+
+			Settings = JsonConvert.DeserializeObject<IrcServerOptions> (settingsJson);
+
+			_log.LogInformation ("Settings Loaded");
+		}
+
+		async Task<string> DownloadIrcConfig (Uri uri)
+		{
```

Comment:
@trixtur Here is the code to download a configuration file. `WebClient` is dated and is replaced by httpclient